lib/setupHooks/remapPathPrefixHook.sh

```diff
@@ -0,0 +1,53 @@
+configureRustcRemapPathPrefix() {
+  local remapTo="${1:-${src:?not defined}}"
+  local remapFrom="${2:-$(pwd)}"
+
+  echo "configuring CARGO_BUILD_RUSTFLAGS to remap paths from \"${remapFrom}\" to \"${remapTo}\""
+
+  # NB: we don't want to directly remap to $src, since this will cause excess
+  # rebuilds stemming from CARGO_BUILD_RUSTFLAGS-related build fingerprint
+  # changes, as buildDepsOnly will be invoked with $src pointing to a `dummySrc`
+  # output`. Instead, we first "neuter" the path to a common intermediate by
+  # stripping the hash part of the path, which we fix up later.
+  local remapToHash=$(echo "$remapTo" | grep --only-matching '@storeDir@/[a-z0-9]\{32\}')
+  if [ "${3:-neuter}" == "neuter" ] && [ -n "$remapToHash" ]; then
+    postBuildHooks+=("fixupNeuteredRustcRemapPathPrefix \"${remapTo}\"")
```

Comment:
Should this instead be run as a `preInstall` hook (or rather any time after checks have run, but if checks are disabled, `preInstall` is the next default phase)?

Basically what I'm wondering is whether re-patching the artifacts could lead to cargo deciding (now or in the future) to suddenly recompile the crate for the sake of testing?